flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java

```diff
@@ -690,7 +683,7 @@ public void testRestartWithSlotSharingAndNotEnoughResources() throws Exception {
 			eg.scheduleForExecution();
 
 			// wait until no more changes happen
-			while (eg.getNumberOfFullRestarts() < numRestarts) {
```

Comment:
You are right. This wait loop is not needed any more since the scheduling and failover happens in the main thread. 
Will replace it with an assertion.

Reply:
done.